services/repository/lfs.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repository
+
+import (
+	"context"
+	"fmt"
+
+	"code.gitea.io/gitea/models/db"
+	git_model "code.gitea.io/gitea/models/git"
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/lfs"
+	"code.gitea.io/gitea/modules/log"
+
+	"xorm.io/builder"
+)
+
+func GarbageCollectLFSMetaObjects(ctx context.Context, logger log.Logger, autofix bool) error {
+	log.Trace("Doing: GarbageCollectLFSMetaObjects")
+
+	if err := db.Iterate(
+		ctx,
+		builder.And(builder.Gt{"id": 0}),
+		func(ctx context.Context, repo *repo_model.Repository) error {
+			return GarbageCollectLFSMetaObjectsForRepo(ctx, repo, logger, autofix)
+		},
+	); err != nil {
+		return err
+	}
+
+	log.Trace("Finished: GarbageCollectLFSMetaObjects")
+	return nil
+}
+
+func GarbageCollectLFSMetaObjectsForRepo(ctx context.Context, repo *repo_model.Repository, logger log.Logger, autofix bool) error {
+	if logger != nil {
```

Comment:
I think it is unnecessary to check  `logger != nil`, it can't be nil, and other checks in `docker` didn't do that.